src/cpu/x64/jit_uni_reorder.cpp

```diff
@@ -1281,12 +1618,164 @@ struct jit_uni_reorder_t : public primitive_t {
+
+            DEBUG({
+                printf("init : ");
+                prb_dump(prb);
+            });
+            // Sort the prb array in increasing sizes of the output stride
+            prb_normalize(prb);
+            DEBUG({
+                printf("norm : ");
+                prb_dump(prb);
+            });
+            /* Combine the variables, which appear together on both
+             * sides of the reorder */
+            prb_simplify(prb);
+            DEBUG({
+                printf("smpl : ");
+                prb_dump(prb);
+            });
+            prb_tile_normalize(prb);
+            DEBUG({
+                printf("smpl : ");
+                prb_dump(prb);
+            });
+
+            if (!tr::jit_single_blk_kernel::applicable(prb)) {
+                return status::unimplemented;
+            }
+
+            auto _pd = new pd_t(attr, src_engine->kind(), src_md,
+                    dst_engine->kind(), dst_md);
+            if (_pd == nullptr) return status::out_of_memory;
+            if (_pd->init(engine, src_engine, dst_engine) != status::success) {
+                delete _pd;
+                return status::unimplemented;
+            }
+            _pd->prb_ = prb;
+            _pd->init_scratchpad_md();
+            return safe_ptr_assign<reorder_pd_t>(*reorder_pd, _pd);
+        }
+
+        tr::prb_t prb_;
+    };
+
+    jit_blk_reorder_t(const pd_t *apd) : primitive_t(apd) {
+        kernel_ = utils::make_unique<tr::jit_single_blk_kernel>(pd()->prb_);
+    }
+
+    inline int n(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].n;
+    }
+    inline int is(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].is;
+    }
+    inline int os(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].os;
+    }
+    inline int ss(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].ss;
+    }
+
+    status_t execute(const exec_ctx_t &ctx) const override {
+        auto in = CTX_IN_MEM(const char *, DNNL_ARG_FROM);
+        auto out = CTX_OUT_MEM(char *, DNNL_ARG_TO);
+        DEFINE_SCALES_BUFFER(scales);
+
+        // kernel handle 2-dimension tiles
+        // Which means there is a tail
+        auto &prb = this->pd()->prb_;
+        ptrdiff_t BH = 1;
+        for (int i = 2; i < prb.ndims; ++i) {
+            BH *= prb.nodes[i].n;
+        }
+
+        auto block_sz = n(0);
+        auto FL = (n(1) + block_sz - 1) / block_sz;
+        auto bh_stride = BH == 1 ? 0 /* don't care, and no is(2) */ : is(2);
+
+        auto itype_sz = data_type_size(pd()->prb_.itype);
+        auto otype_sz = data_type_size(pd()->prb_.otype);
+
+        if (kernel_->has_tail()) {
+            const size_t work_amount = (size_t)BH * FL;
+            int nthr = adjust_num_threads(dnnl_get_max_threads(), work_amount);
+            if (nthr)
```

Comment:
could nthr be equal to 0?

Reply:
It was from paralell_nd template and probably did someone so something like reorder a 1x16x0x0 tensor. I recall we had problem in processing zero tensors.

Reply:
Hmm, interesting... `1x16x0x0` should not perform any reorder in the first place :)
Maybe this is a missing part for all the jitted reorder we have (all one, yeah).
Typically, in the rest of the library we have a check on zero sizes (e.g. see [here](https://github.com/oneapi-src/oneDNN/blob/e5bc59afd372b8ff877cb833c67a623c7029692c/src/cpu/x64/jit_avx2_1x1_convolution.hpp#L73)), and if so, dispatch to the reference implementation. It gracefully does nothing (except that sometimes you need to zero the diff tensors).
I think, you need to add the same check in `pd::init()` or `is_applicable()` function to skip such singularities.